app/uk/gov/hmrc/ivorchestration/repository/IvSessionDataRepositoryAlgebra.scala

```diff
@@ -14,32 +14,33 @@
  * limitations under the License.
  */
 
-package uk.gov.hmrc.ivorchestration.services
+package uk.gov.hmrc.ivorchestration.repository
 
 import reactivemongo.api.indexes.Index
 import reactivemongo.api.indexes.IndexType.Ascending
 import reactivemongo.bson.{BSONDocument, BSONInteger, BSONObjectID}
 import reactivemongo.core.errors.DatabaseException
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.ivorchestration.config.MongoConfiguration
-import uk.gov.hmrc.ivorchestration.model.AuthRetrieval._
-import uk.gov.hmrc.ivorchestration.model.{AuthRetrievalCore, UnexpectedState}
+import uk.gov.hmrc.ivorchestration.model.api.IvSessionData._
+import uk.gov.hmrc.ivorchestration.model.core.{IvSessionDataCore, JourneyId}
+import uk.gov.hmrc.ivorchestration.model.UnexpectedState
 import uk.gov.hmrc.ivorchestration.persistence.DBConnector
 import uk.gov.hmrc.mongo.ReactiveRepository
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
 
-trait AuthRetrievalAlgebra[F[_]] {
-  def insertAuthRetrieval(authRetrievalCore: AuthRetrievalCore)(implicit hc: HeaderCarrier): F[AuthRetrievalCore]
-  def findAuthRetrievals()(implicit hc: HeaderCarrier): F[List[AuthRetrievalCore]]
-  def findJourneyIdAndCredId(journeyId: String, credId: String)(implicit hc: HeaderCarrier): F[Option[AuthRetrievalCore]]
+trait IvSessionDataRepositoryAlgebra[F[_]] {
+  def insertIvSessionData(authRetrievalCore: IvSessionDataCore): F[IvSessionDataCore]
+  def retrieveAll(): F[List[IvSessionDataCore]]
+  def findByKey(journeyId: JourneyId, credId: String): F[Option[IvSessionDataCore]]
 }
 
-class AuthRetrievalDBService(mongoComponent: DBConnector)
-  extends ReactiveRepository[AuthRetrievalCore, BSONObjectID]("authretrievals", mongoComponent.mongoConnector.db, AuthRetrievalCore.format)
-    with AuthRetrievalAlgebra[Future] with MongoConfiguration {
+class IvSessionDataRepositoryDBService(mongoComponent: DBConnector)
```

Comment:
does this need the suffix DBService it seems a bit long how about IvSessionDataRepository

Reply:
renamed